protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -142,17 +139,17 @@ impl<'a, T: Serialize + GetSize + Deserialize<'a>, B: IsBuffer + AeadBuffer> Wit
                 }
                 self.sv2_buffer.danger_set_start(0);
                 let src = self.sv2_buffer.get_data_owned();
-                let frame = Sv2Frame::<T, B::Slice>::from_bytes_unchecked(src);
+                let frame = Sv2Frame::<T, B::Slice>::from_bytes(src)?;
```

Comment:
we can use unchecked here, if you are concerned you can put a check in debug mode